gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+        assertNull(current.dnsName());
+        assertNotNull(current.description());
+        assertNull(current.nameServerSet());
+        assertTrue(zone.nameServers().isEmpty());
+        assertNotNull(current.generatedId());
       }
     } finally {
       DNS.delete(ZONE1.name());
       DNS.delete(ZONE_EMPTY_DESCRIPTION.name());
     }
   }
 
-  // todo(mderka) implement tests for other batch calls, issue #874
+  @Test
+  public void testCreateValidZoneBatch() {
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Zone> completeZoneResult = batch.createZone(ZONE1);
+      DnsBatchResult<Zone> partialZoneResult = batch.createZone(ZONE_EMPTY_DESCRIPTION);
+      batch.submit();
+      Zone created = completeZoneResult.get();
+      assertEquals(ZONE1.description(), created.description());
+      assertEquals(ZONE1.dnsName(), created.dnsName());
+      assertEquals(ZONE1.name(), created.name());
+      assertNotNull(created.creationTimeMillis());
+      assertNotNull(created.nameServers());
+      assertNull(created.nameServerSet());
+      assertNotNull(created.generatedId());
+      Zone retrieved = DNS.getZone(ZONE1.name());
+      assertEquals(created, retrieved);
+      created = partialZoneResult.get();
+      assertEquals(ZONE_EMPTY_DESCRIPTION.description(), created.description());
+      assertEquals(ZONE_EMPTY_DESCRIPTION.dnsName(), created.dnsName());
+      assertEquals(ZONE_EMPTY_DESCRIPTION.name(), created.name());
+      assertNotNull(created.creationTimeMillis());
+      assertNotNull(created.nameServers());
+      assertNull(created.nameServerSet());
+      assertNotNull(created.generatedId());
+      retrieved = DNS.getZone(ZONE_EMPTY_DESCRIPTION.name());
+      assertEquals(created, retrieved);
+    } finally {
+      DNS.delete(ZONE1.name());
+      DNS.delete(ZONE_EMPTY_DESCRIPTION.name());
+    }
+  }
+
+  @Test
+  public void testCreateZoneWithErrorsBatch() {
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Zone> nameErrorResult = batch.createZone(ZONE_NAME_ERROR);
+      DnsBatchResult<Zone> noPeriodResult = batch.createZone(ZONE_DNS_NO_PERIOD);
+      batch.submit();
+      try {
+        nameErrorResult.get();
+        fail("Zone name is missing a period. The service returns an error.");
+      } catch (DnsException ex) {
+        // expected
+        assertFalse(ex.retryable());
+      }
+      try {
+        noPeriodResult.get();
+        fail("Zone name is missing a period. The service returns an error.");
+      } catch (DnsException ex) {
+        // expected
+        assertFalse(ex.retryable());
+      }
+    } finally {
+      DNS.delete(ZONE_NAME_ERROR.name());
+      DNS.delete(ZONE_DNS_NO_PERIOD.name());
+    }
+  }
+
+  @Test
+  public void testCreateZoneWithOptionsBatch() {
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Zone> batchResult =
+          batch.createZone(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.CREATION_TIME));
```

Comment:
Nit: you could import the options from Dns (e.g. Dns.ZoneOption) and remove "Dns." everywhere you use options.  Up to you.